Uzair-Imtiaz/Code-Book
# Logic for authentication added.

### Description
This pull request marks the initial contribution to the project and introduces several key features to the authentication app. The changes include the addition of four templates and modifications to the **views.py** file, aimed at enhancing the user registration, login, and welcome functionalities of the Django application.

### Templates

1. **register.html**: This template serves as the user registration form, allowing users to input their registration details.

2. login.html: The login template presents a form for users to enter their login credentials.

3. **welcome.html**: This template displays a welcome message to authenticated users, providing a warm greeting upon successful login.

4. **base.html**: The base template establishes the layout for other templates by providing a common structure and elements that are extended by the remaining templates.

### Views
1. **RegisterView**: This class-based view manages the user registration process. It includes both the get() and post() methods to render the registration form and handle form submissions, respectively.

2. **LoginView**: Similar to the RegisterView, the LoginView is a class-based view responsible for user login functionality. It provides the get() and post() methods to handle rendering the login form and processing login submissions.

3. **welcome_view**: This function-based view is responsible for rendering the welcome message to authenticated users. It includes only the get() method, allowing users to receive a personalized welcome upon successful login.

### Testing
To test the newly added features and ensure their proper functioning, follow the steps below:

1. Clone the repository:
`git clone git@github.com:Uzair-Imtiaz/Code-Book.git`

2. Create a virtual environment:
`python3 -m venv myenv`

3. Activate the venv:
`source myenv/bin/activate`

4. Install Django:
`pip install Django`

2. Navigate to the project directory:
`cd Code-Book`

6. Apply database migrations:
`python3 manage.py migrate`

7. Start the development server:
`python3 manage.py runserver`

8. Open a web browser and visit the application at **http://localhost:8000/**.

9. Test the user registration functionality by accessing the registration form, entering valid registration details, and verifying that a new user is successfully created.

10. Test the user login functionality by accessing the login form, entering valid login credentials, and confirming that the user is logged in and redirected to the appropriate page.

11. Check the welcome message functionality by logging in as an authenticated user and ensuring that the welcome message is displayed on the relevant page.

### Snapshots
<img width="1272" alt="Screenshot 2023-07-05 at 11 44 24 AM" src="https://github.com/Uzair-Imtiaz/Code-Book/assets/107175205/a72d3096-6c17-49e5-884b-1c56fb44b148">

<img width="1237" alt="image" src="https://github.com/Uzair-Imtiaz/Code-Book/assets/107175205/e7b4ed78-3edb-4e6d-8d0a-f3909e3b6eeb">

<img width="1272" alt="Screenshot 2023-07-05 at 11 51 21 AM" src="https://github.com/Uzair-Imtiaz/Code-Book/assets/107175205/e7ada63c-0f59-4774-aaf6-82026576e4f7">

<img width="1272" alt="Screenshot 2023-07-05 at 11 51 55 AM" src="https://github.com/Uzair-Imtiaz/Code-Book/assets/107175205/13424c56-c5b4-4541-a396-6671e4637ba1">
